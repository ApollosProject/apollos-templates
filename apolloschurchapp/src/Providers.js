import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers, NavigationService } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import {
  LiveProvider,
  ACCEPT_FOLLOW_REQUEST,
} from '@apollosproject/ui-connected';
import { checkOnboardingStatusAndNavigate } from '@apollosproject/ui-onboarding';
import { ONBOARDING_VERSION } from './ui/Onboarding';

import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      // TODO deprecated prop
      navigate={NavigationService.navigate}
      handleExternalLink={(url) => {
        const path = url.split('app-link/')[1];
        const [route, location] = path.split('/');
        if (route === 'content')
          NavigationService.navigate('ContentSingle', { itemId: location });
        if (route === 'nav')
          NavigationService.navigate(
            // turns "home" into "Home"
            location[0].toUpperCase() + location.substring(1)
          );
      }}
      actionMap={{
        // accept a follow request when someone taps "accept" in a follow request push notification
        acceptFollowRequest: ({ requestPersonId }) =>
          client.mutate({
            mutation: ACCEPT_FOLLOW_REQUEST,
            variables: { personId: requestPersonId },
          }),
      }}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() =>
          checkOnboardingStatusAndNavigate({
            client,
            navigation: NavigationService,
            latestOnboardingVersion: ONBOARDING_VERSION,
          })
        }
      >
        <AnalyticsProvider>
          <LiveProvider>
            <Providers
              themeInput={customTheme}
              iconInput={customIcons}
              {...props}
            />
          </LiveProvider>
        </AnalyticsProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
