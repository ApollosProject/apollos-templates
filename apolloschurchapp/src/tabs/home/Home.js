import React, { PureComponent } from 'react';
import { Query } from 'react-apollo';
import { Image } from 'react-native';
import SafeAreaView from 'react-native-safe-area-view';
import { get } from 'lodash';
import PropTypes from 'prop-types';

import {
  ContentCardConnected,
  fetchMoreResolver,
} from '@apollosproject/ui-connected';
import {
  styled,
  FeedView,
  BackgroundView,
  TouchableScale,
  FeaturedCard,
} from '@apollosproject/ui-kit';

import Features from './Features';
import GET_USER_FEED from './getUserFeed';
import GET_CAMPAIGN_CONTENT_ITEM from './getCampaignContentItem';

const LogoTitle = styled(({ theme }) => ({
  height: theme.sizing.baseUnit,
  margin: theme.sizing.baseUnit,
  alignSelf: 'center',
  resizeMode: 'contain',
}))(Image);

class Home extends PureComponent {
  static navigationOptions = () => ({
    header: null,
  });

  static propTypes = {
    navigation: PropTypes.shape({
      getParam: PropTypes.func,
      setParams: PropTypes.func,
      navigate: PropTypes.func,
    }),
  };

  handleOnPress = (item) =>
    this.props.navigation.navigate('ContentSingle', {
      itemId: item.id,
      transitionKey: item.transitionKey,
    });

  render() {
    return (
      <BackgroundView>
        <SafeAreaView>
          <Query
            query={GET_USER_FEED}
            variables={{
              first: 10,
              after: null,
            }}
            fetchPolicy="cache-and-network"
          >
            {({ loading, error, data, refetch, fetchMore, variables }) => (
              <FeedView
                ListItemComponent={ContentCardConnected}
                content={get(data, 'userFeed.edges', []).map(
                  (edge) => edge.node
                )}
                fetchMore={fetchMoreResolver({
                  collectionName: 'userFeed',
                  fetchMore,
                  variables,
                  data,
                })}
                isLoading={loading}
                error={error}
                refetch={refetch}
                ListHeaderComponent={
                  <>
                    <LogoTitle source={require('./wordmark.png')} />
                    <Query
                      query={GET_CAMPAIGN_CONTENT_ITEM}
                      fetchPolicy="cache-and-network"
                    >
                      {({ data: featuredData, loading: isFeaturedLoading }) => {
                        const featuredContent = get(
                          featuredData,
                          'campaigns.edges',
                          []
                        ).map((edge) => edge.node);

                        const featuredItem = get(
                          featuredContent[0],
                          'childContentItemsConnection.edges[0].node',
                          {}
                        );

                        return (
                          <TouchableScale
                            onPress={() =>
                              this.handleOnPress({ id: featuredItem.id })
                            }
                          >
                            <ContentCardConnected
                              Component={FeaturedCard}
                              contentId={featuredItem.id}
                              isLoading={isFeaturedLoading}
                            />
                          </TouchableScale>
                        );
                      }}
                    </Query>
                    <Features navigation={this.props.navigation} />
                  </>
                }
                onPressItem={this.handleOnPress}
              />
            )}
          </Query>
        </SafeAreaView>
      </BackgroundView>
    );
  }
}

export default Home;
