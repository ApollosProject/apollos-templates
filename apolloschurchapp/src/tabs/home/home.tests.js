import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../Providers';

import Home from './Home';
import GET_USER_FEED from './getUserFeed';

describe('User Home Feed Query', () => {
  it('should return correct query results', () => {
    const mock = {
      request: {
        query: GET_USER_FEED,
      },
      result: {
        data: {
          userFeed: {
            edges: [
              {
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:678b4a38968fc6004dd8b23e586c923e',
                  coverImage: {
                    name: 'Square Image',
                    sources: [
                      {
                        uri:
                          'https://apollosrock.newspring.cc/GetImage.ashx?guid=ed857076-2623-4fdc-8476-50b3a60e0b68',
                      },
                    ],
                  },
                  parentChannel: {
                    id: 'ContentChannel:7fba1b1ee253e0fd5f0795b4b8b1175e',
                    name: 'Devotion Series',
                    iconName: 'text',
                  },
                  title: 'Psalms: A 28-Day Devotional',
                },
              },
              {
                node: {
                  __typename: 'UniversalContentItem',
                  id: 'UniversalContentItem:0f361c619b7e5dd511da181069498250',
                  coverImage: {
                    name: 'Square Image',
                    sources: [
                      {
                        uri:
                          'https://apollosrock.newspring.cc/GetImage.ashx?guid=aa68b674-9cf4-4f8a-8702-9564a4a9fa7b',
                      },
                    ],
                  },
                  parentChannel: {
                    id: 'ContentChannel:7fba1b1ee253e0fd5f0795b4b8b1175e',
                    name: 'Devotion Series',
                    iconName: 'text',
                  },
                  title: '2 Samuel: A 5-Week Devotional',
                },
              },
            ],
          },
        },
      },
    };

    const navigation = {
      getParam: jest.fn(),
      setParams: jest.fn(),
      navigate: jest.fn(),
    };
    const tree = renderer.create(
      <Providers mocks={[mock]} addTypename={false}>
        <Home navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
