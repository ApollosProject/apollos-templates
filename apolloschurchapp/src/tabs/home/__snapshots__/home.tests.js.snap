// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Home Feed Query should return correct query results 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294967295,
      4294506484,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={
        <mapProps(getContext(mapProps(withPropsOnChange(mapProps(Image)))))
          source={
            Object {
              "testUri": "../../../src/tabs/home/wordmark.png",
            }
          }
        />
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      loading={true}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={0.7}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/tabs/home/wordmark.png",
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 16,
                "margin": 16,
                "resizeMode": "contain",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</BVLinearGradient>
`;
